From: Sam Ravnborg <sam@ravnborg.org>
Date: Mon, 21 May 2012 20:45:37 +0200
Subject: [PATCH] net: drop NET dependency from HAVE_BPF_JIT

commit e47b65b032f2997aa0a7392ecdf656c86d4d7561 upstream.

There is no point having the NET dependency on the select target, as it
forces all users to depend on NET to tell they support BPF_JIT.  Move
the config option to the bottom of the file - this could be a nice place
also for future "selectable" config symbols.

Fix up all users to drop the dependency on NET now that it is not
required to supress warnings for non-NET builds.

Reported-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
Acked-by: David Miller <davem@davemloft.net>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 arch/arm/Kconfig     |    2 +-
 arch/powerpc/Kconfig |    2 +-
 arch/sparc/Kconfig   |    2 +-
 arch/x86/Kconfig     |    2 +-
 net/Kconfig          |    7 ++++---
 5 files changed, 8 insertions(+), 7 deletions(-)

Index: linux/arch/arm/Kconfig
===================================================================
--- linux.orig/arch/arm/Kconfig	2012-06-24 23:41:24.000000000 +0200
+++ linux/arch/arm/Kconfig	2012-06-24 23:49:03.000000000 +0200
@@ -30,7 +30,7 @@
 	select HAVE_SPARSE_IRQ
 	select GENERIC_IRQ_SHOW
 	select CPU_PM if (SUSPEND || CPU_IDLE)
-	select HAVE_BPF_JIT if NET
+	select HAVE_BPF_JIT
 	help
 	  The ARM series is a line of low-power-consumption RISC chip designs
 	  licensed by ARM Ltd and targeted at embedded applications and
Index: linux/arch/powerpc/Kconfig
===================================================================
--- linux.orig/arch/powerpc/Kconfig	2012-06-20 00:18:30.000000000 +0200
+++ linux/arch/powerpc/Kconfig	2012-06-24 23:49:03.000000000 +0200
@@ -134,7 +134,7 @@
 	select GENERIC_IRQ_SHOW_LEVEL
 	select HAVE_RCU_TABLE_FREE if SMP
 	select HAVE_SYSCALL_TRACEPOINTS
-	select HAVE_BPF_JIT if (PPC64 && NET)
+	select HAVE_BPF_JIT if PPC64
 	select HAVE_ARCH_JUMP_LABEL
 	select ARCH_HAVE_NMI_SAFE_CMPXCHG
 
Index: linux/arch/x86/Kconfig
===================================================================
--- linux.orig/arch/x86/Kconfig	2012-06-23 17:09:51.000000000 +0200
+++ linux/arch/x86/Kconfig	2012-06-24 23:49:03.000000000 +0200
@@ -72,7 +72,7 @@
 	select GENERIC_CLOCKEVENTS_MIN_ADJUST
 	select IRQ_FORCED_THREADING
 	select USE_GENERIC_SMP_HELPERS if SMP
-	select HAVE_BPF_JIT if (X86_64 && NET)
+	select HAVE_BPF_JIT if X86_64
 	select CLKEVT_I8253
 	select ARCH_HAVE_NMI_SAFE_CMPXCHG
 
Index: linux/net/Kconfig
===================================================================
--- linux.orig/net/Kconfig	2012-06-20 00:18:30.000000000 +0200
+++ linux/net/Kconfig	2012-06-24 23:49:03.000000000 +0200
@@ -232,9 +232,6 @@
 	depends on SMP && SYSFS && USE_GENERIC_SMP_HELPERS
 	default y
 
-config HAVE_BPF_JIT
-	bool
-
 config BPF_JIT
 	bool "enable BPF Just In Time compiler"
 	depends on HAVE_BPF_JIT
@@ -326,3 +323,7 @@
 
 
 endif   # if NET
+
+# Used by archs to tell that they support BPF_JIT
+config HAVE_BPF_JIT
+	bool
